internal/services/keyvault/key_vault_key_resource.go

```diff
@@ -698,6 +697,10 @@ func expandKeyVaultKeyOptions(d *pluginsdk.ResourceData) *[]keyvault.JSONWebKeyO
 }
 
 func expandKeyVaultKeyRotationPolicy(v interface{}) keyvault.KeyRotationPolicy {
+	if v == nil || len(v.([]interface{})) == 0 {
```

Comment:
which means this can become:

```suggestion
func expandKeyVaultKeyRotationPolicy(v []interface{}) keyvault.KeyRotationPolicy {
	if len(v) == 0 {
```